src/tests/simple_pipeline.c

```diff
@@ -0,0 +1,307 @@
+/**
+ * (C) Copyright 2020-2021 Intel Corporation.
+ *
+ * SPDK-License-Identifier: BSD-2-Clause-Patent
+ */
+
+/*
+ * This provides a simple example for how to use the data filter capabilities
+ * in DAOS.
+ */
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <sys/types.h>
+#include <daos.h>
+
+/** datos info */
+static daos_handle_t	poh; // pool
+static daos_handle_t	coh; // container
+static daos_handle_t	oh;  // object
+
+#define	ASSERT(cond, ...)					\
+do {								\
+	if (!(cond)) {						\
+		fprintf(stderr, __VA_ARGS__);			\
+		exit(1);					\
+	}							\
+} while (0)
+
+/** DB info */
+#define NR_IODS		3
+#define STRING_LEN	10
+static char*	fields[NR_IODS]	= {"owner", "species", "sex"};
+
+
+void
+insert_example_records (void)
+{
+	int		rc;
+	d_iov_t		dkey;
+	d_sg_list_t	sgls[NR_IODS];
+	d_iov_t		iovs[NR_IODS];
+	daos_iod_t	iods[NR_IODS];
+	uint32_t	i, j;
+
+	char	name[8][STRING_LEN]   		= {"Slim\0",
+						   "Buffy\0",
+						   "Claws\0",
+						   "Whistler\0",
+						   "Chirpy\0",
+						   "Browser\0",
+						   "Fang\0",
+						   "Fluffy\0"};
+	char	owner[8][STRING_LEN]   		= {"Benny\0",
+						   "Harold\0",
+						   "GWen\0",
+						   "Gwen\0",
+						   "Gwen\0",
+						   "Diane\0",
+						   "Benny\0",
+						   "Harold\0"};
+	char	species[8][STRING_LEN] 		= {"snake\0",
+						   "dog\0",
+						   "cat\0",
+						   "bird\0",
+						   "bird\0",
+						   "dog\0",
+						   "dog\0",
+						   "cat\0"};
+	char	sex[8][STRING_LEN]  		= {"m\0",
+						   "f\0",
+						   "m\0",
+						   "m\0",
+						   "f\0",
+						   "m\0",
+						   "m\0",
+						   "f\0"};
+	char	**data[NR_IODS];
+
+	data[0] = (char **)owner;
+	data[1] = (char **)species;
+	data[2] = (char **)sex;
+
+	for (i = 0; i < 8; i++) { /** records */
+		/** set dkey for record */
+		d_iov_set(&dkey, name[i], STRING_LEN);
+		for (j = 0; j < NR_IODS; j++) { /** fields */
+			/** akeys */
+			sgls[j].sg_nr		= 1;
+			sgls[j].sg_nr_out	= 0;
+			sgls[j].sg_iovs		= &iovs[0];
+			d_iov_set(&iovs[0], data[j][i], STRING_LEN);
+
+			d_iov_set(&iods[j].iod_name, (void *)fields[j],
+					strlen(fields[j]));
+			iods[j].iod_nr		= 1;
+			iods[j].iod_size	= STRING_LEN;
+			iods[j].iod_recxs	= NULL;
+			iods[j].iod_type	= DAOS_IOD_SINGLE;
+		}
+		rc = daos_obj_update(oh,DAOS_TX_NONE,0,&dkey,NR_IODS,iods,sgls,NULL);
```

Comment:
(style)  space required after that ',' (ctx:VxV)
(style)  space required after that ',' (ctx:VxV)
(style)  space required after that ',' (ctx:VxO)
(style)  space required before that '&' (ctx:OxV)
(style)  space required after that ',' (ctx:VxV)
(style)  space required after that ',' (ctx:VxV)
(style)  space required after that ',' (ctx:VxV)
(style)  space required after that ',' (ctx:VxV)